docs/index.md

```diff
@@ -1,27 +1,264 @@
+    limit: 10,
+    desc: true
+});
+
+// Select the last run of your Actor that finished
+// with a SUCCEEDED status.
+const lastSucceededRunClient = actorClient.lastRun({ status: 'SUCCEEDED' });
+// Fetches items from the run's dataset.
+const { items } = await lastSucceededRunClient.dataset()
+    .listItems();
+```
+
+The quick access to `dataset` and other storage directly from the run client can be used with the [`lastRun()`](https://docs.apify.com/api/client/js/reference/class/ActorClient#lastRun) method.
+
+## Features
+
+Based on the endpoint, the client automatically extracts the relevant data and returns it in the expected format. Date strings are automatically converted to `Date` objects. For exceptions, the client throws an [`ApifyApiError`](https://docs.apify.com/api/client/js/reference/class/ApifyApiError), which wraps the plain JSON errors returned by API and enriches them with other contexts for easier debugging.
+
+```js
+const { ApifyClient } = require('apify-client');
+
+const apifyClient = new ApifyClient({ token: 'MY-APIFY-TOKEN' });
+
+try {
+   const { items } = await client.dataset("non-existing-dataset-id").listItems();
+} catch (error) {
+   // The error is an instance of ApifyApiError
+   const { message, type, statusCode, clientMethod, path } = error;
+   // Log error for easier debugging 
+   console.log({ message, statusCode, clientMethod, type });
+}
+```
+
+### Retries with [exponential backoff](https://docs.apify.com/api/client/js/docs/features#retries-with-exponential-backoff)
+
+Network communication sometimes fails. That's a given. The client will automatically retry requests that failed due to a network error, an internal error of the Apify API (HTTP 500+), or a rate limit error (HTTP 429). By default, it will retry up to 8 times. The first retry will be attempted after ~500ms, the second after ~1000ms, and so on. You can configure those parameters using the `maxRetries` and `minDelayBetweenRetriesMillis` options of the `ApifyClient` constructor.
+
+```js
+const { ApifyClient } = require('apify-client');
+
+const apifyClient = new ApifyClient({ 
+  token: 'MY-APIFY-TOKEN',
+  maxRetries: 8,
+  minDelayBetweenRetriesMillis: 500, // 0.5s
+  timeoutSecs: 360 // 6 mins
+});
+```
+
+### Convenience functions and options
+
+Some actions can't be performed by the API itself, such as indefinite waiting for an Actor run to finish (because of network timeouts). The client provides convenient `call()` and `waitForFinish()` functions that do that. If the limit is reached, the returned promise is resolved to a run object that will have status `READY` or `RUNNING` and it will not contain the Actor run output.
+
+[Key-value store](https://docs.apify.com/platform/storage/key-value-store) records can be retrieved as objects, buffers, or streams via the respective options, dataset items can be fetched as individual objects or serialized data and we plan to add better stream support and async iterators.
```

Comment:
Please do not mention future plans in documentation.

```suggestion
[Key-value store](https://docs.apify.com/platform/storage/key-value-store) records can be retrieved as objects, buffers, or streams via the respective options, dataset items can be fetched as individual objects or serialized data.
```

Reply:
That was part of the content we already have, I also wanted to remove it 👍